source/csharp/Camera/Basic/CaptureViaZivid/CaptureViaZivid.cs

```diff
@@ -97,29 +109,43 @@ private static HalconDotNet.HObjectModel3D ZividToHalconPointCloud(Zivid.NET.Poi
         {
             for (uint j = 0; j < width; j++)
             {
-                double x = pointsXYZ[i, j, 0];
+                float x = pointsXYZ[i, j, 0];
+                float normal = normalsXYZ[i, j, 0];
```

Comment:
Since you replaced these with float, which is correct since Zivid returns float iirc, then I think using `float.IsNaN(x)` in other places is correct, right?